File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/gui/FrmCompararDiretorioProgresso.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 08/07/2015
 * Time: 11:24
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using HFSGuardaDiretorio.comum;

namespace HFSGuardaDiretorio.gui
{
	/// <summary>
	/// Description of FrmCompararDiretorioProgresso.
	/// </summary>
	public class FrmCompararDiretorioProgresso : IProgressoLog
	{
    	FrmCompararDiretorio form;

    	public FrmCompararDiretorioProgresso(FrmCompararDiretorio form) 
    	{
        	this.form = form;
		}
    	
		public void ProgressoLog(Progresso progresso)
		{
	        if (progresso.Posicao == 0) {
	            this.form.pb.Minimum = progresso.Minimo;
	            this.form.pb.Maximum = progresso.Maximo;
	            this.form.pb.Step = progresso.Passo;
	        }
	        this.form.pb.Value = progresso.Posicao;
		}
    	
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/gui/FrmCompararDiretorioProgresso.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 08/07/2015
' * Time: 11:24
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports HFSGuardaDiretorio.comum

Namespace gui
	''' <summary>
	''' Description of FrmCompararDiretorioProgresso.
	''' </summary>
	Public Class FrmCompararDiretorioProgresso
		Implements IProgressoLog
		Private form As FrmCompararDiretorio

		Public Sub New(form As FrmCompararDiretorio)
			Me.form = form
		End Sub

		Public Sub ProgressoLog(progresso As Progresso) Implements IProgressoLog.ProgressoLog
			If progresso.Posicao = 0 Then
				Me.form.pb.Minimum = progresso.Minimo
				Me.form.pb.Maximum = progresso.Maximo
				Me.form.pb.[Step] = progresso.Passo
			End If
			Me.form.pb.Value = progresso.Posicao
		End Sub

	End Class
End Namespace
